CI: use anchors for paths

pjonsson opened the pull request:
The path lists are meant to be identical,
so use the new yaml anchors to ensure that.

<!-- readthedocs-preview datacube-explorer start -->
----
📚 Documentation preview 📚: https://datacube-explorer--941.org.readthedocs.build/en/941/

<!-- readthedocs-preview datacube-explorer end -->

pjonsson committed:
CI: use anchors for paths

The path lists are meant to be identical,
so use the new yaml anchors to ensure that.

omad reviewed (commented):
I'm not 100% sure the cognitive overhead of YAML anchors is worth it here...

_(30s pause to read it again)_ 

Can't we delete those path lines entirely? Isn't `**` the default?

pjonsson committed:
CI: conditionally run tests

Only trigger the CI workflows when
something they test has changed. This
avoids spawning slow jobs when they
are not required.

pjonsson commented:
I'm not sure about the default, but the CI jobs in this repository are quite slow so I went ahead and added a bunch of paths so the jobs are only triggered when something that they test has changed.
